crates/polars-time/src/upsample.rs

```diff
@@ -122,59 +122,65 @@ fn upsample_impl(
 ) -> PolarsResult<DataFrame> {
     let s = source.column(index_column)?;
     let time_type = s.dtype();
-    if matches!(time_type, DataType::Date) {
```

Comment:
the diff in this file is a little hard to parse, could you summarise the gist of the change please?

i'm not sure I see why `upsample_impl` needs to change